bftengine/src/bftengine/BFTEngine.cpp

```diff
@@ -1,93 +1,44 @@
-//Concord
+// Concord
 //
-//Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
 //
-//This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in compliance with the Apache 2.0 License. 
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
 //
-//This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
 
 #include "Replica.hpp"
 #include "ReplicaImp.hpp"
 
-namespace bftEngine
-{
-	namespace impl
-	{
-		struct ReplicaInternal : public Replica
```

Comment:
This trivial class/wrapper was deliberately created to separate the implementation from the external interface. This class simplifies the code. It is used to encapsulate some implementations details (for example, sequence number are represented by different types). It also shows that we don’t recommend to downcast Replica to ReplicaImp. At some point, it may also help to split ReplicaImp into several classes/modules.

Reply:
I understand that may have been the intention. However, practically speaking, this separation makes it difficult to add new functionality to the ReplicaImp and expose it. In order to add a system wide aggregator, I need a way to add that aggregator to the ReplicaImp. I can add another parameter to `Replica::createNewReplica`, but that forces all users to care about the aggregator, even though we are only using it inside tests right now. 

Can you please suggest a mechanism for adding the aggregator functionality to the ReplicaImp that abides by your implementation choices?

Reply:
Nevermind, I suppose I can just add the method to ReplicaInternal. That will work fine.